### DISCUSSION TITLE
How to handle Spatie Media Library items?

### DISCUSSION THREAD

**Discussion by goellner** [spatie/laravel-data#285] [Q&A]
Is anyone including media library items in their DTO and has an example for me how you set this up? When returning the content as JSON we need to include the media items in the response.

**Comment by Nir-An** [comment]
Hi @goellner 

I don't familiar with the package and I need more information 
but if I understand you get a collection so maybe  these things can be useful: 

the type of [getMedia](https://github.com/spatie/laravel-medialibrary/blob/a7d1542c4fa6d8f46ce7132718599030c568f36d/src/InteractsWithMedia.php#L267)

```php

return type : MediaCollections\Models\Collections\MediaCollection

```
1. try to change the collection to ``array`` and just add the array to the object


```php
...
public function __construct(
     ...
    public array $media
}
public function fromModel(ModelWithMedia $model)
{
   return new self(){
    ...
   media: $model->getMedia->toArray()
    ...
}

```
2. and then try to add it as a collection.

```php

...
 public function __construct(
...
     public MediaCollection $media

...
}
public function fromModel(ModelWithMedia $model)
{
return new self{
...
media: $model->getMedia
...
}
}
